hal/src/rtl872x/ble_hal.cpp

```diff
@@ -2980,6 +2980,10 @@ ssize_t BleGatt::notifyValue(hal_ble_attr_handle_t attrHandle, const uint8_t* bu
                 }
                 uint8_t retry = 3;
                 bool success;
+                if (BleEventDispatcher::getInstance().isThreadCurrent()) {
```

Comment:
Nit: might look better if we just get out of the do/while loop below based on `isThreadCurrent()` ?